src/redash_stmo/health.py

```diff
@@ -0,0 +1,108 @@
+import os
+import json
+import time
+import redash
+
+from redash_stmo import settings
+
+from flask import jsonify
+from flask_login import login_required
+
+from celery.utils.log import get_task_logger
+from redash import models, redis_connection, statsd_client
+from redash.worker import celery
+from redash.utils import parse_human_time
+from redash.monitor import get_status as original_get_status
+from redash.query_runner import BaseQueryRunner
+from redash.handlers.base import routes
+from redash.permissions import require_super_admin
+
+logger = get_task_logger(__name__)
+
+
+def store_health_status(data_source_id, data_source_name, query_text, data):
+    key = "data_sources:health"
+
+    cache = json.loads(redis_connection.get(key) or '{}')
+    if data_source_id not in cache:
+        cache[data_source_id] = {
+            "metadata": {"name": data_source_name},
+            "queries": {}
+        }
+    cache[data_source_id]["queries"][query_text] = data
+
+    cache[data_source_id]["status"] = "SUCCESS"
+    for query_status in cache[data_source_id]["queries"].values():
+        if query_status["status"] == "FAIL":
+            cache[data_source_id]["status"] = "FAIL"
+            break
+
+    redis_connection.set(key, json.dumps(cache))
+
+
+@celery.task(name="redash_stmo.health.update_health_status", time_limit=90, soft_time_limit=60)
+def update_health_status():
+    for data_source in models.DataSource.query:
```

Comment:
So Redash supports different "organizations" and data sources can be assigned to them individually (the Mozilla setup only has one organization, the default one). There is a notion of a "current organization" per request (e.g. /org1/api/.. and /org2/api/..) but since this task is run outside the request cycle, we can't refer to [`current_org`](https://github.com/getredash/redash/blob/25de0b9dab4096711b1d38400b0b4c5a3d03b458/redash/authentication/org_resolving.py#L19) as Redash does all over the place of the request time code. I don't have a suggestion, but just wanted you to be aware that the status API will mix data sources that could be configured to be separate for different organizations users. It's probably a good idea to ask Arik about it when we move this feature upstream.